adapters/play/src/main/scala/caliban/Uploads.scala

```diff
@@ -0,0 +1,128 @@
+package caliban
+
+import java.nio.file.{ Files, Path }
+
+import caliban.InputValue.ListValue
+import caliban.Uploads.{ FileMeta, Uploads }
+import caliban.Value.{ NullValue, StringValue }
+import caliban.schema.Annotations.GQLName
+import zio.blocking.Blocking
+import zio.stream.{ Stream, ZSink, ZStream }
+import zio.{ Chunk, Has, Layer, RIO, UIO, URIO, ZIO, ZLayer }
+
+@GQLName("Upload")
+sealed case class Upload(name: String) {
```

Comment:
sealed forbids inheritance outside of this file, final forbids overriding in case of inheritance. do we want users at all to extend this case class?

Reply:
https://stackoverflow.com/a/31133245/2521865

IIRC `sealed` is mainly for `trait` hierarchies because they can be used by the compiler to generate exhaustivity checks and `final` can have some useful properties during compilation (though I don't know if that would apply to such a simple case class).

Reply:
Let it be final then, I don't have all the information either to make a perfect decision.